### ISSUE TITLE
Error with toml package when deploying project

### ISSUE THREAD

**Issue by alfonsomhc** [Miserlou/Zappa#1246]
I get this error when I deploy my project:
```
Packaging project as gzipped tarball.
  Could not find a version that satisfies the requirement toml==0.9.3 (from versions: 0.6.0, 0.6.5, 0.7.0, 0.7.1, 0.8.0, 0.8.1, 0.8.2, 0.9.0, 0.9.1, 0.9.2, 0.9.3.1)
No matching distribution found for toml==0.9.3
```
My understanding is that when zappa gets installed, the requirement is  toml==0.9.3, but when it tries to make the package that will be send to Amazon, it does not find toml==0.9.3 (it does find toml==0.9.3.1 though).

Would it be a solution to update the requirements.txt (and require toml==0.9.3.1?

My environment is python 3.6, and the latest zappa available in pip.

By the way I think the deployment should fail in this kind of situation. I guess this is what #1235 is about.

**Comment by alfonsomhc** [comment]
I tried to install that package version:
```
$ pip install toml==0.9.3.1
Collecting toml==0.9.3.1
Installing collected packages: toml
  Found existing installation: toml 0.9.3
    Uninstalling toml-0.9.3:
      Successfully uninstalled toml-0.9.3
Successfully installed toml-0.9.3
```

I checked in pypi (https://pypi.python.org/pypi/toml) and it seems that in fact the two version packages are the same.

Therefore I don't think requirements.txt needs to be udpated. It's probably a bug in the packaging system that doesn't support four levels in the version?

**Comment by jwkvam** [comment]
I think the same issue is causing recent problems in this issue as well https://github.com/Miserlou/Zappa/issues/64

**Comment by mcrowson** [comment]
Would this be fixed if Zappa didn't peg requirement versions?

**Comment by Miserlou** [comment]
This is `toml`'s problem, pinned requirements or not.

**Comment by mcrowson** [comment]
but it would allow the user to pin the working version of toml and still use the latest/greatest zappa

**Comment by tycoles** [comment]
Sorry to bother as I'm not at all familiar with maintaining widely used open source projects.

I basically forced zappa to use toml==0.9.3.1 and `zappa deploy` worked fine. I'm confused though, because zappa uses `toml>=0.9.3`. Wouldn't the available version `0.9.3.1` fulfill that criteria?

**Comment by tycoles** [comment]
So it looks like the problem with toml is referenced here:

https://github.com/uiri/toml/issues/129

As it stands, either we wait for a new release (0.9.4) or update requirements to >=0.9.3.1

**Comment by tycoles** [comment]
Sorry if this is an ignorant question, but would it not be possible to update requirements to `toml>=0.9.3.1`? Otherwise I think we're stuck waiting for toml's version to increment for this to be fixed.

**Comment by tycoles** [comment]
Yeah it looks like we either roll back to toml 0.9.2 (which may not be possible) or wait for a new toml release (0.9.4). Zappa version 0.44.3 is the latest version of Zappa with working toml (0.9.2) in it, in case anyone out there is wondering.
